src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -1397,6 +1403,50 @@ methodHandle SharedRuntime::find_callee_method(TRAPS) {
   return callee_method;
 }
 
+JRT_BLOCK_ENTRY(void, SharedRuntime::end_training_check_c1(JavaThread* current))
+{
+  JRT_BLOCK
+    SharedRuntime::end_training_check(CHECK);
+  JRT_BLOCK_END
+}
+JRT_END
+
+JRT_BLOCK_ENTRY(void, SharedRuntime::end_training_check_c2(JavaThread* current))
+{
```

Comment:
this and the _c1 method don't need to check the method flag to see if its a 'trigger', this is because we checked that on code generation which predicates the call to these methods